src/ResourceManager/Common/Commands.Common.Strategies/StateExtensions.cs

```diff
@@ -70,7 +70,19 @@ sealed class ContainsDispatchVisitor : IEntityConfigVisitor<IState, bool>
         {
             public bool Visit<TModel>(ResourceConfig<TModel> config, IState context)
                 where TModel : class
-                => context.Contains(config);
+            {
+                TModel currendEntityConfig = context.Get(config);
```

Comment:
This's a generic algorithm. It supposes only check if the config exist. It shouldn't validate anything else. I would propose to create a separate algorithm with different name.

Reply:
@sergey-shandar The implementation of the validation logic is generic as well, its the particular resource that can choose to override the validation logic, the default is true for all resources. As of now only the LB resource that implements the validation.

The purpose of the check is to see if the existing config is compatible with what we are targeting. The code path here is such that it have both... Infact, the existing logic checks if the curent config exists. This should be a great place to add the validate... it a logical extension of the check to see if a config already exists to add the logic to now see if it is compatible with the target config for the resource. Do let me know why you think the current changes are not generic...